examples/01_mnist_template.py

```diff
@@ -0,0 +1,182 @@
+from __future__ import print_function
+import argparse
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+import torch.optim as optim
+from torchvision import datasets, transforms
+from torch.optim.lr_scheduler import StepLR
+
+###### Hydra Block ######
+from typing import List, Any
+from omegaconf import MISSING
+import hydra
+from hydra.core.config_store import ConfigStore
+from dataclasses import dataclass
+
+# config schema imports
+from config.torch.optim import AdadeltaConf
+from config.torch.optim.lr_scheduler import StepLRConf
+
+@dataclass
+class ExportedArgparseArgs:
+    epochs: int = 14
+    batch_size: int = 64
+    test_batch_size: int = 1000
+    no_cuda: bool = False
+    save_model: bool = False
+    dry_run: bool = False
+    log_interval: int = 10
+    seed: int = 1
+    checkpoint_name: str = 'unnamed.pt'
+
+@dataclass
+class Conv2dConf:
+    in_channels: int = MISSING 
+    out_channels: int = MISSING 
+    kernel_size: int = 3
+    stride: int = 1
+
+@dataclass
+class DropoutConf:
+    p: float = MISSING
+
+@dataclass
+class Maxpool2dConf:
+    kernel_size: int = 2
+    stride: Any = None 
+    padding: int = 0
+    dilation: int = 1
+
+@dataclass
+class LinearConf:
+    in_features: int = MISSING
+    out_features: int = MISSING 
+
+@dataclass
+class MNISTNetConf:
+    conv1: Conv2dConf = Conv2dConf(in_channels=1, out_channels=32)
+    conv2: Conv2dConf = Conv2dConf(in_channels=32, out_channels=64)
+    dropout1: DropoutConf = DropoutConf(p=0.25)
+    dropout2: DropoutConf = DropoutConf(p=0.5)
+    maxpool: Maxpool2dConf = Maxpool2dConf()
+    linear1: LinearConf = LinearConf(in_features=9216, out_features=128)
+    linear2: LinearConf = LinearConf(in_features=128, out_features=10)
+
+@dataclass
```

Comment:
I have some configs for torch vision in NeMo:
https://github.com/NVIDIA/NeMo/blob/main-vis-res/nemo/collections/cv/datasets/configs.py

for transforms:
 - https://github.com/NVIDIA/NeMo/blob/main-vis-res/nemo/collections/cv/transforms/configs.py

I can add them in the next iteration

Reply:
Definitely! Since you started working on the dataset portion of the project, I figured I'd work on all the other components first and give you some time =].

We should be able to generate these dataset configs using configen as well. Shall we set that up and create a PR? At least we can mirror the vision datasets in `torchvision`. I guess for now `transform` and `target transform` have to be passthrough args.

Reply:
Once this is done, I will add them to the Intermediate Tutorial (there will be a separate PR for that soon) 🙂